### PR TITLE
Add a page on large world coordinates (double-precision floats)

### PR DESCRIPTION
**Pull request by Calinou**
**Note:** This currently doesn't include the SCons option rename from https://github.com/godotengine/godot/pull/67399.

### CODE REVIEW

File: tutorials/physics/large_world_coordinates.rst [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+|                      |                       |                       | past this value.                                                            |
++----------------------+-----------------------+-----------------------+-----------------------------------------------------------------------------+
+
+When using single-precision floats, it is possible to go past the suggested
+ranges, but more visible artifacting will occur and physics glitches will be
+more common (such as the player not walking straight in certain directions).
+
+.. seealso::
+
+    See the `Demistifying Floating Point Precision <https://blog.demofox.org/2017/11/21/>`__
```
Comment (mhilbrunner) [inline tutorials/physics/large_world_coordinates.rst:107]:
```suggestion
    See the `Demystifying Floating Point Precision <https://blog.demofox.org/2017/11/21/>`__
```

File: tutorials/physics/large_world_coordinates.rst [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+rendering engine will increase its precision for rendering operations when large
+world coordinates are enabled. The shaders do not use double-precision floats
+for performance reasons, but an `alternative solution <https://github.com/godotengine/godot/pull/66178>`__
+is used to emulate double precision for rendering using single-precision floats.
+
+.. note::
+
+    Enabling large world coordinates comes with a performance and memory usage
+    penalty, especially on 32-bit CPUs. Only enable large world coordinates if
+    you actually need it.
```
Comment (mhilbrunner) [inline tutorials/physics/large_world_coordinates.rst:136]:
```suggestion
    you actually need them.
```
Or `need to`, I guess?

**Review by mhilbrunner — COMMENTED** [review]
Great content! Could maybe mention the shifting-world-origin alternative approach that many games use, as thats a common workaround.

**Comment by Calinou** [comment]
I mentioned origin shifting in passing, but I don't know how to implement it. It can quickly get complicated (especially for multiplayer games), so it's mainly good to know for low-end mobile.

I added a paragraph on resource saving and interoperability as well.

**Comment by mhilbrunner** [comment]
> I mentioned origin shifting in passing

D'oh, yeah, thats enough already :)

> I don't know how to implement it

Roughly: whenever player distance to origin exceeds the shift treshold, calculate the difference between player position and origin and shift _everything's_ position in the game by that difference, so the player is at the origin again, but everything else is still at the same position relative to each other.

Then, pile lots of bandaids and fixes on top as you run into different bugs and edge cases :P

Those additional questions are good ones, no idea honestly, may try out the GDExtension thing. Regardless, I'll go ahead and merge this one now, and we can amend and iterate as needed if we figure out more :) Amazing work, as usual! 🎉

**Review by mhilbrunner — APPROVED** [review]

**Pull request merged**